torchaudio/datasets/speechcommands.py

```diff
@@ -131,6 +132,27 @@ def __init__(
             walker = sorted(str(p) for p in Path(self._path).glob("*/*.wav"))
             self._walker = [w for w in walker if HASH_DIVIDER in w and EXCEPT_FOLDER not in w]
 
+    def get_metadata(self, n: int) -> Tuple[str, int, str, str, int]:
+        """Get metadata for the n-th sample from the dataset. Returns filepath instead of waveform,
+        but otherwise returns the same fields as :py:func:`__getitem__`.
+
+        Args:
+            n (int): The index of the sample to be loaded
+
+        Returns:
+            (str, int, str, str, int):
+            ``(filepath, sample_rate, label, speaker_id, utterance_number)``
+        """
+        fileid = self._walker[n]
+        return _get_speechcommands_metadata(fileid, self._archive)
+
+    def _load_waveform(self, path: str):
+        path = os.path.join(self._archive, path)
+        waveform, sample_rate = torchaudio.load(path)
+        if sample_rate != SAMPLE_RATE:
```

Comment:
The way `SAMPLE_RATE` global variable is used looks a bit strange to me. It should be used only once in `get_metadata`, and load function should be using the value returned from `get_metadata` function, instead of referring to the global value again.